src/main/java/com/example/demo1/controller/api/ShoppingCartApiController.java

```diff
@@ -0,0 +1,74 @@
+package com.example.demo1.controller.api;
+
+import com.example.demo1.model.dto.ShoppingCartDto;
+import com.example.demo1.model.dto.ShoppingCartReturnDto;
+import com.example.demo1.model.dto.ShoppingCartUpdateDto;
+import com.example.demo1.service.ShoppingCartService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/shopping-cart")
+@RequiredArgsConstructor
+public class ShoppingCartApiController {
+    private final ShoppingCartService shoppingCartService;
+    //create ShoppingCart
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public void createShoppingCart(@RequestBody ShoppingCartDto shoppingCartDto){
+        shoppingCartService.create(shoppingCartDto);
+    }
+
+    //read ShoppingCart
+    @GetMapping
+    public List<ShoppingCartReturnDto> getShoppingCart(){
+        return shoppingCartService.findAll();
+    }
+
+    @GetMapping("/description/{description}")
+    public List<ShoppingCartReturnDto> getShoppingCartByDescription
```

Comment:
shopping carts are a bit special, you kinda want to set a cookie on the users' browser to identify his shopping cart id so that every time he comes to your website he has the products already in the cart. Maybe you want to implement something like this

Reply:
also don't take long strings as path variables it use request param better or request body. it will make your urls like this:

http://mysite.com/also%20don't%20take%20long%20strings%20as